### ISSUE TITLE
The DockerHub publish action bump demo race condition

### ISSUE THREAD

**Issue by slominskir** [JeffersonLab/container-workflows#1]
If additional commits are pushed while the publish action is running and already done a checkout, but hasn't reached the push step then the bump demo version will fail because a pull must be done first.  This can happen if you update the README while a CD workflow is running.

https://github.com/JeffersonLab/container-workflows/blob/2000de75007f901053c4aedf822986378f047d56/.github/workflows/docker-publish.yml

Example log:

```
Run git config --local user.name "github-actions[bot]"
From https://github.com/JeffersonLab/cnm
 * branch            main       -> FETCH_HEAD
   41de3a5..40f2a7d  main       -> origin/main
Already on 'main'
Your branch is behind 'origin/main' by 1 commit, and can be fast-forwarded.
  (use "git pull" to update your local branch)
[main b567a3f] Bump demo compose version to 2.8.2
 1 file changed, 1 insertion(+), 1 deletion(-)
To https://github.com/JeffersonLab/cnm
 ! [rejected]        main -> main (non-fast-forward)
error: failed to push some refs to 'https://github.com/JeffersonLab/cnm'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. If you want to integrate the remote changes,
hint: use 'git pull' before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

Will be a race condition even if we add a pull.    Might need to just avoid doing commits during a release/deployment, which seems reasonable.    Also consider using "latest" tag in DockerHub instead of having to update the version reference each release.  The latest tag isn't without it's own issues , but may make sense in this case.

**Comment by slominskir** [comment]
Fixed in [v2.0.0](https://github.com/JeffersonLab/container-workflows/releases/tag/v2.0.0)
